instana/instrumentation/boto3_inst.py

```diff
@@ -8,14 +8,25 @@
 import inspect
 
 from ..log import logger
-from ..singletons import tracer
+from ..singletons import tracer, agent
 from ..util.traceutils import get_active_tracer
 
 try:
     import opentracing as ot
     import boto3
     from boto3.s3 import inject
 
+    def extract_custom_headers(span, headers):
+        if agent.options.extra_http_headers is None:
```

Comment:
Just to be safe I would also add `or headers is None`. Unlikely to happen, but theoretically still possible.